pkg/workloads/console/runner/runner.go

```diff
@@ -736,10 +769,56 @@ func (c *Runner) GetAttachablePod(csl *workloadsv1alpha1.Console) (*corev1.Pod,
 	}
 
 	for _, c := range pod.Spec.Containers {
-		if c.TTY {
+		if !csl.Spec.Interactive || c.TTY {
 			return pod, c.Name, nil
 		}
 	}
 
 	return nil, "", errors.New("no attachable pod found")
 }
+
+// NoninteractiveAttacher knows how to attach to stdio of an existing container, relaying io
+// to the parent process file descriptors.
+type NoninteractiveAttacher struct {
```

Comment:
Can we shift this up in the file to sit just under the `InteractiveAttacher` struct and funcs?